presto-docs/src/main/sphinx/optimizer/reorder-joins.rst

```diff
@@ -63,8 +65,19 @@ For the following query:
 
 .. code-block:: sql
 
-  SELECT * FROM part p, orders o, lineitem l, supplier s, nation n
-  WHERE p.partkey = l.partkey AND l.orderkey = o.orderkey AND l.suppkey = s.suppkey AND s.nationkey = n.nationkey;
+ SELECT * FROM part p, orders o, lineitem l, supplier s, nation n
+ WHERE p.partkey = l.partkey AND l.orderkey = o.orderkey AND l.suppkey = s.suppkey AND s.nationkey = n.nationkey;
 
 The join order will change from ``part, orders, lineitem, supplier, nation`` to
 ``part, lineitem, orders, supplier, nation``.
+
+
+COST_BASED
+==========
+
+When the join reordering strategy is set to ``COST_BASED``, Presto will use statistics
+provided by the connectors to estimate the costs for different join orders (currently stats
+are only available in the hive connector). It will then choose the join order with the lowest
+computed cost. If statistics are not available or if for any other reason a cost could not be
+computed, the ``ELIMINATE_CROSS_JOINS`` strategy is used instead. Enumerating all possible join
+orders for many tables is computationally intensive, so joins are reordered in groups of ten at a time.
```

Comment:
Could you explain what happens if a query has more than 10 joins, say 12? Would 10 first joins be reordered between them and then the remaining 2 will be reordered separately? and the clusters of 10 and 2 will be merged without considering any cost between the two groups?